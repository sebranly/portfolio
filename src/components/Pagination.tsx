import * as React from 'react';
import classnames from 'classnames';

export interface PaginationProps {
  allPages: number[];
  onClickPage: (nb: number) => void;
  pageNumber: number;
}

const Pagination: React.FC<PaginationProps> = (props) => {
  const { allPages, onClickPage, pageNumber } = props;

  return (
    <div className="flex w-full">
      {allPages.map((page: number, index: number) => {
        const isSelected = pageNumber === page;
        const classnamesTab = classnames(
          {
            'bg-black': isSelected,
            'text-white': isSelected,
            'rounded-bl-lg': isSelected && index !== 0,
            'rounded-br-lg': isSelected && index !== allPages.length - 1
          },
          `p-2 flex-1`
        );

        return (
          <button
            className={classnamesTab}
            key={page}
            style={{ fontWeight: isSelected ? 'bold' : 'normal' }}
            type="submit"
            onClick={() => onClickPage(page)}
          >
            {page}
          </button>
        );
      })}
    </div>
  );
};

export { Pagination };
