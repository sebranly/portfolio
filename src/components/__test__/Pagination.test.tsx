import * as React from 'react';
import { render } from '@testing-library/react';

import { Pagination } from '../Pagination';

test('Pagination', () => {
  const createComponent = () => <Pagination allPages={[1, 2]} onClickPage={jest.fn()} pageNumber={2} />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
