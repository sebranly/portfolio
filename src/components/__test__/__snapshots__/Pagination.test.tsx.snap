// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination 1`] = `
NodeList [
  <div
    class="flex w-full"
  >
    <button
      class="p-2 flex-1"
      style="font-weight: normal;"
      type="submit"
    >
      1
    </button>
    <button
      class="bg-black text-white rounded-bl-lg p-2 flex-1"
      style="font-weight: bold;"
      type="submit"
    >
      2
    </button>
  </div>,
]
`;
